### PR TITLE
update vm based installation

### PR DESCRIPTION
**Pull request by princerachit**
Signed-off-by: Shovan Maity <shovan.cse91@gmail.com>

### CODE REVIEW

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc [hunk 1]
```diff
@@ -1,32 +1,39 @@
-= On-premise example
-
-[abstract]
-CMOS is deployed to monitor an on-premise cluster with a simple Nginx web server exposed on port 8080 to provide a landing page.
-
-include::partial$tutorial.adoc[]
+= CMOS On Premise Deployment
```
Comment (princerachit) [inline docs/modules/ROOT/pages/tutorial-onpremise.adoc]: Add an overview section about CMOS here. In the second paragraph of overview say something like:
```txt
CMOS can also be deployed on a VM based infrastructure, this is achieved by effectively running the docker container
```

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc [hunk 2]
```diff
@@ -1,32 +1,39 @@
...
 .Microlith overview
 image::microlith-runtime.png[]
 
-== Usage
+CMOS is deployed to monitor an on-premise cluster with a simple Nginx web server exposed on port 8080 to provide a landing page.
 
-To run refer to the xref:quickstart.adoc[quick start guide] for full details:
+== Deployment
 
+At this moment we only support deployment using docker. You can get started by deploying the docker container:
```
Comment (princerachit) [inline docs/modules/ROOT/pages/tutorial-onpremise.adoc]: === Accessing the deployed services
There are two ways to access the services deployed in the container
==== Landing page
A landing page provided by the cmos stack through nginx has links to all the deployed services. You can navigate to <host_ip>:8080 to access the landing page.
Attach screenshot here with highlight to the links to services

==== Direct port access
You can directly access the service via exposed ports either through browser or through any other consuming service. e.g. to access grafana which is exposed on port 3000 you can navigate to <ip>:3000 in your browser.

In most of the cases you might endup setting port forwarding or port exposuire within your network, exposing ports of the runing docker container has no special steps and you can follow the general guidelines used in you r organisation.

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
+[source, console]
+----
+$ docker container port cmos 8080
+----
+----
+0.0.0.0:49278
+:::49278
+----
 
 == Configuration options
```
Comment (princerachit) [inline docs/modules/ROOT/pages/tutorial-onpremise.adoc:35]: CMOS gives you fine grained control to the various services included. <Diagram reference> .. As shown in the diagram there are some default values ... We provide ways to override these values using environment variables and native configuration files of each component.

Continue with what is already there.

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc [hunk 4]
```diff
@@ -133,6 +140,21 @@ The cluster manager will auto-start and configure itself with credentials suppli
 
 The cluster manager exposes its REST API from the container so this can be used externally to add/remove Couchbase clusters. We also support running any scripts found in `/etc/healthcheck` to do it.
 
+=== Couchbase Cluster
```
Comment (princerachit) [inline docs/modules/ROOT/pages/tutorial-onpremise.adoc]: Let us pull this at root level which is at the same level as that of deployment and title it as:
== Connecting Couchbase
After you have deployed...is to connect the cluster to actuaqlly observe the cluster.
Two ways:

Verification:

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc (hunk 3 above)
Comment (princerachit) [reply docs/modules/ROOT/pages/tutorial-onpremise.adoc:35]: Also, Get rid of single big snippet code about environment variable configuration and put it into each component section.

**Review by princerachit — COMMENTED** [review]
Please address as discussed over call and comments. Make sure you check that proper rendering is being done.

File: docs/modules/ROOT/pages/tutorial-onpremise.adoc [hunk 5]
```diff
@@ -101,37 +111,111 @@ To add Couchbase Server end points, create a similar JSON format file to the lin
...
+COUCHBASE_PWD=password
+COUCHBASE_ENDPOINT=http://<hostname/IP>:8091
+
+# Make the api call via curl. You can use any other tool to make this API call
+curl -u "${CLUSTER_MONITOR_USER}:${CLUSTER_MONITOR_PWD}" -X POST -d '{ "user": "'"${COUCHBASE_USER}"'", "password": "'"${COUCHBASE_PWD}"'", "host": "'"${COUCHBASE_ENDPOINT}"'" }' "${CLUSTER_MONITOR_ENDPOINT}/api/v1/clusters"
+----
+
+== Configure Grafana [[configure_grafana]]
+Various dashboard to monitor couchbase cluster is shipped out of the box in Grafana. Connect a couchbase cluster and navigate to Grafana to see the graphs. You can list all the dashboards using the https://grafana.com/docs/grafana/latest/dashboards/search/#dashboard-search[search dashboard option^]. You may create https://grafana.com/docs/grafana/latest/getting-started/getting-started/#step-3-create-a-dashboard[additional dashboards^] as per your needs.
+
```
Comment (princerachit) [inline docs/modules/ROOT/pages/tutorial-onpremise.adoc:178]: @shovanmaity can you add a screen shot here. Saying something like:
```
Post complete setup you should be able to see metrics in your grafana dashboard:
<Attach a screenshot here>
```

And the same for Alerting rules, we need a screenshot of alert manager page.

Comment (shovanmaity) [reply docs/modules/ROOT/pages/tutorial-onpremise.adoc:178]: done

**Pull request closed without merging**